lib/exception_handling/escalate_callback.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'escalate'
+
+module ExceptionHandling
```

Comment:
👏 This is super clean! I love it! Something to make sure of is we currently in `web` and `invoca_microservice` have a registration of another `on_escalate` that is doing `honeybadger` escalations. We should make sure when updating those services, to remove this `on_escalate` so that we avoid double escalating exceptions to honeybadger.  This would actually be a really good demo for this ticket:

1. remove the on_escalate from invoca_microservice/web
2. update invoca_microservice/web to the latest version of this gem
3. show that when an exception occurs within `Invoca::ServiceDiscovery` that the exception is still logged and sent to honeybadger

Reply:
@jebentier Good point re: `invoca_microservice`. I'll update that.

BTW it just occurred to me that I may have introduced a race condition. I have `require 'exception_handling' triggering its registration for `on_escalate`. There's a chance that that callback could get triggered before `exception_handling` has been configured. We have a `rescue` block around the whole thing so it wouldn't be terrible. But it would be a bit less buggy to not do the registration until configuration is complete... or at least ignore the callback until then. It would make sense to be part of this block in `invoca_microservice`, but of course that doesn't always run:
```
    def setup_exception_handling(service_root, logger)
      filter_list_filename                   = File.expand_path("config/exception_filters.yml", service_root)
      ExceptionHandling.eventmachine_safe    = true
      ExceptionHandling.logger               = logger
      ExceptionHandling.filter_list_filename = filter_list_filename
      ExceptionHandling.server_name          = Invoca::Env.service_hostname
      ExceptionHandling.post_log_error_hook  = method(:post_exception_metrics)
      if defined?(Rails) && Rails.respond_to?(:backtrace_cleaner)
        Rails.backtrace_cleaner.remove_silencers!
      end
    end
```
Maybe just make the callback a no-op when there's no `ExceptionHandling.logger =` assigned? In practice that's the most important configuration that we need. (If we had block-based config I could wait for that to complete. But we don't right now.)

Worth doing anything, or just count on the `rescue` to make the race condition pretty painless?

Reply:
@jebentier Also, I looked in `web` and I'm not finding the `on_escalate`? Are you sure it's there? Possibly in a branch?

Reply:
I think it's important to at least throw out to `STDERR` if the logger isn't set up yet. Would that be at all possible to do? And in terms of the web `on_escalate` it's that it uses [invoca_microservices here](https://github.com/Invoca/web/blob/master/config/application.rb#L256) for the setup. So updating `invoca_microservices` should do the trick.

Reply:
Good point about STDERR. That does make it tempting just to defer this registration until the logger is set, since that's exactly what will happen when there's nothing registered.

Since the registration is idempotent, I don't need to check for it. I think I just need the registration 2 places:

1. when first required, register if logger set
2. in `logger =`, register

Sound good?

And good to hear that there's no special callback in `web`.